UPSTREAM-MERGE.sh

```diff
@@ -133,6 +128,16 @@ git commit -m "Merge upstream tag $version" -m "Operator SDK $version" -m "Merge
 # verify merge is correct
 git --no-pager log --oneline "$(git merge-base origin/"$rebase_branch" tags/"$version")"..tags/"$version"
 
+# update vendor directory
+go mod tidy
+go mod vendor
+# capture $? of go mod vendor, if it fails skip the add
+git add vendor
+# just to make sure that the vendor directory is actually updated 
+git diff --staged --quiet && { echo "No changed files in vendor directory?! Aborting."; exit 1; }
```

Comment:
It sounds like the script will abort even when there is no expected change in `go.mod` and so in the vendor directory. Does that mean we cannot use this script if there is no difference in `go.mod` from upstream?

Reply:
These steps are after the merge commit of the rebase. So the rebase would still work. However, I wanted to give a signal to the user that the vendor directory is not updated and the user has to take care of it manually. That's why the abort message.

Reply:
Yeah, I got that. My only concern is it will _always_ abort even when we are not expecting any vendor file changes.  I think vendor will be updated if there is any modification in `go.mod` from the upstream, which might not happen during every rebase, right?

Reply:
> That may be a case, but I think the chances are less for that scenario. Even for that case, it would be good to alert the user that the vendor directory didn't have any changes so that the user cross-checks it. 

How about writing this way? It would solve both concerns. 

```bash
go mod tidy && go mod vendor || { echo "go mod vendor failed. Aborting!"; exit 1; }
if ! git diff --quiet vendor/; then
  git add vendor
  git commit -m "UPSTREAM: <drop>: Update vendor directory"
else
  echo "No changed files in vendor directory. Skipping add."
fi
```